### PR TITLE
feat(Copy): Support for minimal and icon modes

### PR DESCRIPTION
**Pull request by benjamincharity**
1st commit:
- quick copy icon now appears on hover
- support minimal mode
- support icon mode
- remove scrollbar
- quick copy enabled by default
- updated demo for more control
- using TsTooltip in place of a browser tooltip for content

2nd commit:
- update ToC that was missed in previous PR

example

### CODE REVIEW

File: terminus-ui/copy/src/copy.component.ts [hunk 1]
```diff
@@ -83,7 +111,29 @@ export class TsCopyComponent {
...
+  @Input()
+  public set format(value: TsCopyDisplayFormat) {
+    this._format = value ? value : 'standard';
+
+    if (this.format === 'icon' && !this.enableQuickCopy) {
+      this.enableQuickCopy = true;
+
+      // istanbul ignore else
+      if (isDevMode()) {
+        throw new TsUILibraryError(`'enableQuickCopy' must be set to 'true' when using the icon only display mode`);
```
Comment (atlwendy) [inline terminus-ui/copy/src/copy.component.ts:128]: `TsUILibraryError`! Nice

File: terminus-ui/utilities/src/error/ui-error.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/**
+ * A Terminus UI specific Error
+ */
+export class TsUILibraryError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.message = message;
+    this.name = 'TsUILibraryError';
+    this.stack = (new Error(message)).stack;
```
Comment (atlwendy) [inline terminus-ui/utilities/src/error/ui-error.ts:9]: 👍

**Review by atlwendy — APPROVED** [review]

File: terminus-ui/copy/src/copy.component.ts (hunk 1 above)
Comment (benjamincharity) [reply terminus-ui/copy/src/copy.component.ts:128]: Yeah! Originally created it for two reasons,
a) It's nice for the stack trace to call out our library specifically when debugging
b) I had read about a way to create a more useful stack trace - which I couldn't get working initially but hope to revisit.

Ended up still using for `a)` at the very least

Comment (benjamincharity) [reply terminus-ui/copy/src/copy.component.ts:128]: Only real difference in use is that we can do `throw Error(...)` but with ours we _must_ call new `throw new TsUILibraryError(...)`

**Pull request merged**